deploy/kubeconfiggenerator.py

```diff
@@ -488,62 +488,56 @@ def flatten(yaml_contents, flattened, types_dict, prefix=''):
-                            #    del types_dict[key]['items']
-                            namelist.append(l['name'])
                         if isinstance(l, dict):
                             inner_prop_dict = {}
                             prop_dict = {'properties': inner_prop_dict}
                             prop_dict['type'] = 'object'
-                            if 'name' in l:
-                                types_dict[key]['properties'][l['name']] = prop_dict
-                            else:
-                                types_dict[key]['items'] = prop_dict
-                                types_dict[key]['type'] = 'array'
+                            types_dict[key]['items'].append(prop_dict)
                         if isinstance(l, list):
                             inner_prop_dict = {}
                             prop_dict = {'items': inner_prop_dict}
                             prop_dict['type'] = 'array'
-                            if 'name' in l:
-                                types_dict[key]['properties'][l['name']] = prop_dict
-                            else:
-                                types_dict[key]['items'] = prop_dict
-                                types_dict[key]['type'] = 'array'
+                            types_dict[key]['items'].append(prop_dict)
                         flatten(l, flattened, inner_prop_dict, prefix=prefix + key + ".")
+                        if 'name' in l:
+                            prop_list_dict.update(inner_prop_dict)
                     else:
                         flattened[prefix + key] = l
-                if len(namelist) > 0:
-                    for name in namelist:
-                        propNameDict = {}
-                        propNameDict['properties'] = None
-                        propNameDict['required'] = []
-                        propNameDict['required'].append(name)
-                        types_dict[key]['anyOf'].append(propNameDict)
-                    for propName in types_dict[key]['properties']:
-                        del types_dict[key]['properties'][propName]['properties']['name']
+                        if isinstance(l, str):
+                            types_dict[key]['items'].append({'type': 'string'})
+                        if isinstance(l, bool):
+                            types_dict[key]['items'].append({'type': 'boolean'})
+                        elif isinstance(l, int):
+                            types_dict[key]['items'].append({'type': 'integer'})
+                        elif isinstance(l, float):
+                            types_dict[key]['items'].append({'type': 'number'})
+                if types_dict[key]['items']:
+                    if len(prop_list_dict) > 1:
+                        types_dict[key]['items'] = {'type': 'object', 'properties': prop_list_dict, 'required': ['name']}
+                        types_dict[key]['x-kubernetes-list-map-keys'] = ['name']
```

Comment:
Should lines 537 and 538 be outside the if block? Currently, if the len(prop_list_dict) <=1, then the two attributes ('x-kubernetes-list-map-keys') and ('x-kubernetes-list-type') will not be added to the types_dict.